File: 1.supprime.py
import os
import time

log_dir = "/var/log/myapp"
now = time.time()

for filename in os.listdir(log_dir):
    filepath = os.path.join(log_dir, filename)
    if os.path.isfile(filepath) and now - os.path.getmtime(filepath) > 7 * 86400:
        os.remove(filepath)

File: 1.supprime.go
package main

import (
    "os"
    "path/filepath"
    "time"
)

func main() {
    logDir := "/var/log/myapp"
    cutoff := time.Now().AddDate(0, 0, -7)

    filepath.Walk(logDir, func(path string, info os.FileInfo, err error) error {
        if err == nil && !info.IsDir() && info.ModTime().Before(cutoff) {
            os.Remove(path)
        }
        return nil
    })
}